### DISCUSSION TITLE
Unknown MySQL server host 'xinwang' (-3)

### DISCUSSION THREAD

**Discussion by oldman1991** [theory/sqitch#641] [Q&A]
sqitch status db:mysql://xiwang:aqpthyYQ7EaUj#asa^UkFXy%^@database-dev.awas.us-east-5.rds.amazonaws.com:3306/test
return error ，Unknown MySQL server host 'xinwang' (-3)
Is there something wrong with my configuration?

**Comment by theory** [comment]
You're not the first to run into this issue. There are two things going on here:

1. The character `#` starts a comment in most sells, so your URI is probably truncated
2. It's a URI, so you need to escape it

Try this instead:

```
db:mysql://xiwang:aqpthyYQ7EaUj%23asa%5EUkFXy%25%5E@database-dev.awas.us-east-5.rds.amazonaws.com:3306/test
```

Also, consider omitting passwords from the URIs; have a look at [sqitch-authentication](http://sqitch.org/docs/manual/sqitch-authentication/) for details.

**Reply by oldman1991** [reply]
ok, thanks for your answer.
I ran into the next problem when I continued。
<img width="713" alt="image" src="https://user-images.githubusercontent.com/9280199/174461128-a5c17ea2-9bb0-4d90-98e6-a1e825dc70cc.png">
why  there is no change? here is my config:
[core]
	engine = mysql
	# plan_file = sqitch.plan
	# top_dir = .
 \# [engine "mysql"]
	# target = db:mysql:
	# registry = sqitch
	# client = mysql
[target "test"]
	uri = db:mysql://xiwang:aqpthyYQ7EaUj%23asa%5EUkFXy%25%5E@database-dev.awas.us-east-5.rds.amazonaws.com:3306/test
[engine "mysql"]
	target = test

And the corresponding sqitch management table is not generated in my database

**Comment by theory** [comment]
You have to run `sqitch deploy` to deploy your changes. It's not automatic.

**Reply by oldman1991** [reply]
I try it ,but it not work
`root@ip-172-31-5-12:~/sqitch_test# sqitch deploy`
`Unknown database 'sqitch'`
`root@ip-172-31-5-12:~/sqitch_test# sqitch deploy`
`Unknown database 'sqitch'`

**Comment by theory** [comment]
You need to tell it what target to use:

```
sqitch deploy test
```

**Reply by oldman1991** [reply]
thanks for your attention
I tried it, still the same problem
`root@ip-172-31-5-12:~/sqitch_test# sqitch deploy test`
`Unknown database 'sqitch'`

**Reply by theory** [reply]
Hrm. It should be creating that database if it doesn't exist right here:

https://github.com/sqitchers/sqitch/blob/b901fe8ca013d40bcb1edfc245bf9054d7cb7985/lib/App/Sqitch/Engine/mysql.pm#L280-L285

Maybe user `xiwang` doesn't have permission? Would expect to see an error if not, but that's my next guess. If you can't grant `xiwang` permission to create the database, you can create it for that user and make the user the owner.

In general the user used to deploy changes should have pretty broad permissions.

**Reply by oldman1991** [reply]
I used this user to create the database directly through the sql command

So it is authorized


The error message is
`Unknown database 'sqitch'`
So is there something wrong with my configuration?

**Comment by theory** [comment]
Actually this looks like the same issue as reported in #601. Can you try renaming `try_lock` in mysql.pm to something else (like _try_lock). That will disable the locking.

**Reply by oldman1991** [reply]
I am using aws rds， so, what should i do?

**Reply by theory** [reply]
How are you running Sqitch?

**Reply by oldman1991** [reply]
I am using the sqitch installed by the Ubuntu operating system according to the instructions on the official website

**Reply by oldman1991** [reply]
so, what should I do, can you help me?

**Comment by theory** [comment]
Can you rename `try_lock` in `mysql.pm` and try again? You can probably find out where the file is by running `perldoc -l App::Sqitch::Engine::mysql`.

**Comment by oldman1991** [comment]
![image](https://user-images.githubusercontent.com/9280199/176660349-eae4de96-667c-4f3a-ac26-e8bc0b0d21b8.png)
I am sorry

**Reply by theory** [reply]
That's odd. It should show you where th file is:

```
$ perldoc -l App::Sqitch::Engine::mysql
/usr/local/perl/lib/5.36.0/lib/perl5/site_perl/5.36.0/App/Sqitch/Engine/mysql.pm
```

**Reply by theory** [reply]
What is the output of `head $(which sqitch)`?

**Reply by oldman1991** [reply]
> 
`/usr/local/bin/sqitch`

**Reply by theory** [reply]
I think you left out the `head` bit.

**Comment by theory** [comment]
Found the issue; watch #601 for the fix. Sorry for the hassle, and I appreciate your patience!
